phoenix-core/src/test/java/org/apache/phoenix/expression/MathTrigFunctionTest.java

```diff
@@ -0,0 +1,179 @@
+            },
+            {
+                new Long[]{(long) 1, (long) 0, (long) -1, Long.MAX_VALUE,
+                    Long.MIN_VALUE, (long) 123, (long) -123},
+                PLong.INSTANCE
+            },
+            {
+                new Long[]{(long) 1, (long) 0, Long.MAX_VALUE, (long) 123},
+                PUnsignedLong.INSTANCE
+            },
+            {
+                new Integer[]{1, 0, -1, Integer.MAX_VALUE, Integer.MIN_VALUE, 123, -123},
+                PInteger.INSTANCE
+            },
+            {
+                new Integer[]{1, 0, Integer.MAX_VALUE, 123},
+                PUnsignedInt.INSTANCE
+            },
+            {
+                new Short[]{(short) 1, (short) 0, (short) -1, Short.MAX_VALUE,
+                    Short.MIN_VALUE, (short) 123, (short) -123},
+                PSmallint.INSTANCE
+            },
+            {
+                new Short[]{(short) 1, (short) 0, Short.MAX_VALUE, (short) 123},
+                PSmallint.INSTANCE
+            },
+            {
+                new Byte[]{(byte) 1, (byte) 0, (byte) -1, Byte.MAX_VALUE,
+                    Byte.MIN_VALUE, (byte) 123, (byte) -123},
+                PTinyint.INSTANCE
+            },
+            {
+                new Byte[]{(byte) 1, (byte) 0, Byte.MAX_VALUE, (byte) 123},
+                PTinyint.INSTANCE
+            }
+    });
+    }
+
+    private boolean testExpression(LiteralExpression literal, double expectedResult,
+                                          String testedFunction) throws SQLException {
+        List<Expression> expressions = Lists.newArrayList((Expression) literal);
+        ImmutableBytesWritable ptr = new ImmutableBytesWritable();
+        Expression mathFunction = null;
+
+        if (testedFunction.equals("SIN")) {
+            mathFunction = new SinFunction(expressions);
+        } else if (testedFunction.equals("COS")) {
+            mathFunction = new CosFunction(expressions);
+        } else if (testedFunction.equals("TAN")) {
+            mathFunction = new TanFunction(expressions);
+        }
+
+        boolean ret = mathFunction.evaluate(null, ptr);
+        if (ret) {
+            Double result =
+                    (Double) mathFunction.getDataType().toObject(ptr, mathFunction.getSortOrder());
+            assertTrue(BaseTest.twoDoubleEquals(result.doubleValue(), expectedResult));
+        }
+
+        return ret;
+    }
+
+    private void test(Number value, PNumericType dataType, double expectedResult,
+                             String testedFunction)
+            throws SQLException {
+        LiteralExpression literal = LiteralExpression.newConstant(value, dataType, SortOrder.ASC);
+        boolean ret1 = testExpression(literal, expectedResult, testedFunction);
+
+        literal = LiteralExpression.newConstant(value, dataType, SortOrder.DESC);
+        boolean ret2 = testExpression(literal, expectedResult, testedFunction);
+        assertEquals(ret1, ret2);
```

Comment:
Why is this required? If both `ret1` and `ret2` return false, this would still pass. Is that intended?

Reply:
As discussed offline, @yanxinyi will fix this in another Jira.